File: backend-node/src/models/PlayerTelemetryState.ts
export interface PlayerTelemetryState {
    currentTime: number;
    trackId: string;
    song: string;
    artist: string;
    duration: number;
    isPlaying: boolean;
    isBuffering: boolean;
    contentRect?: {
        left: number;
        top: number;
        width: number;
        height: number;
    };
    browserTime: number;
    timestamp: number; // Server time when received
}

File: app/src/main/java/com/example/music_room/data/remote/model/PlayerTelemetryState.kt
package com.example.music_room.data.remote.model

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class PlayerTelemetryState(
    val currentTime: Long,
    val trackId: String,
    val song: String,
    val artist: String,
    val duration: Long,
    val isPlaying: Boolean,
    val isBuffering: Boolean,
    val browserTime: Long
)

@JsonClass(generateAdapter = true)
data class PlayerTickMessage(
    val type: String = "player_tick",
    val payload: PlayerTelemetryState
)

@JsonClass(generateAdapter = true)
data class SocketEventMessage(
    val type: String,
    val payload: Any?
)
